ci-operator/step-registry/azure/provision/custom-role/azure-provision-custom-role-commands.sh

```diff
@@ -0,0 +1,385 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# save the exit code for junit xml file generated in step gather-must-gather
+# pre configuration steps before running installation, exit code 100 if failed,
+# save to install-pre-config-status.txt
+# post check steps after cluster installation, exit code 101 if failed,
+# save to install-post-check-status.txt
+EXIT_CODE=100
+trap 'if [[ "$?" == 0 ]]; then EXIT_CODE=0; fi; echo "${EXIT_CODE}" > "${SHARED_DIR}/install-pre-config-status.txt"' EXIT TERM
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+function run_cmd_with_retries_save_output()
+{
+    local cmd="$1" output="$2" retries="${3:-}"
+    local try=0 ret=0
+    [[ -z ${retries} ]] && max="20" || max=${retries}
+    echo "Trying ${max} times max to run '${cmd}', save output to ${output}"
+
+    eval "${cmd}" > "${output}" || ret=$?
+    while [ X"${ret}" != X"0" ] && [ ${try} -lt ${max} ]; do
+        echo "'${cmd}' did not return success, waiting 60 sec....."
+        sleep 60
+        try=$(( try + 1 ))
+        ret=0
+        eval "${cmd}" > "${output}" || ret=$?
+    done
+    if [ ${try} -eq ${max} ]; then
+        echo "Never succeed or Timeout"
+        return 1
+    fi
+    echo "Succeed"
+    return 0
+}
+
+function create_role_definition_json() {
+
+    local role_name=$1 permissions=$2 role_definition_file=$3
+
+    role_description="the custom role ${role_name} with minimal permissions for cluster ${CLUSTER_NAME}"
+    assignable_scopes="""
+\"/subscriptions/${AZURE_AUTH_SUBSCRIPTOIN_ID}\"
+"""
+
+    # create role definition json file
+    jq --null-input \
+       --arg role_name "${role_name}" \
+       --arg description "${role_description}" \
+       --argjson assignable_scopes "[ ${assignable_scopes} ]" \
+       --argjson permission_list "[ ${permissions} ]" '
+{
+  "Name": $role_name,
+  "IsCustom": true,
+  "Description": $description,
+  "assignableScopes": $assignable_scopes,
+  "Actions": $permission_list,
+  "notActions": [],
+  "dataActions": [],
+  "notDataActions": []
+}' > "${role_definition_file}"
+}
+
+function create_custom_role() {
+    local role_definition="$1"
+
+    # create custom role
+    cmd="az role definition create --role-definition ${role_definition}"
+    run_command "${cmd}" || return 1
+
+    echo "Sleep 1 min to wait for custom role created"
+    sleep 60
+}
+
+if [[ "${AZURE_INSTALL_USE_MINIMAL_PERMISSIONS}" == "no" ]]; then
```

Comment:
We do not need this block any more.

Reply:
Or append `&& [[ "${ENABLE_MIN_PERMISSION_FOR_STS}" == "false" ]]`

Reply:
append `&& [[ "${ENABLE_MIN_PERMISSION_FOR_STS}" == "false" ]] ` to print message that "skip the step when both envs are not set"